src/preamble.js

```diff
@@ -2305,14 +2309,23 @@ function integrateWasmJS(Module) {
 
   // Memory growth integration code
   Module['reallocBuffer'] = function(size) {
-    size = Math.ceil(size / wasmPageSize) * wasmPageSize; // round up to wasm page size
+    var PAGE_MULTIPLE = Module["usingWasm"] ? 65536 : 16777216; // In wasm, heap size must be a multiple of 64KB. In asm.js, they need to be multiples of 16MB.
```

Comment:
in fact both of those constants could have nice names, as they are each used more than once.